api

```diff
@@ -351,6 +351,38 @@ app_to_pkgname() { #given an app-name, convert it to a unique, valid package-nam
   echo "pi-apps-$(echo "$app" | md5sum | cut -c1-8 | awk '{print $1}')"
 }
 
+add_external_repo() { # add an external apt repo and its gpg key
+  repopath="$1"
+  reponame="$2"
+  pubkeyurl="$3"
+
+  # make pi-apps key storage dir if it doesn't exist
+  if [ ! -d /tmp/pi-apps-apt-keys ]; then
+    mkdir -p /tmp/pi-apps-apt-keys || error "add_external_repo: failed to create pi-apps key storage dir."
+  fi
+
+  # check if .list file already exists
+  [ -f /etc/apt/sources.list.d/${reponame}.list ] && error "add_external_repo: /etc/apt/sources.list.d/${reponame}.list already exists."
+
+  # download gpg key from specified url
+  rm -f /tmp/pi-apps-apt-keys/${reponame}-keyring.gpg
+  wget -q "$pubkeyurl" -O /tmp/pi-apps-apt-keys/${reponame}-keyring.gpg || error "add_external_repo: download from specified pubkeyurl failed." && rm -f /tmp/pi-apps-apt-keys/${reponame}-keyring.gpg
```

Comment:
I recommend testing the conditionals in this line.
If wget succeeds, the `||` will be skipped but the `&&` will run `rm`, which immediately removes the file `wget` just downloaded. Is that intended?
Conversely, if wget fails, the `||` will trigger, which runs the `error` function and exit the script. In this case, the `rm` command won't run, when I think you intended it to run.

Reply:
My intention was that if wget fails, then the incomplete download file should be removed as well as the error printed.

I have removed the && part anyways, because the function already checks if the url is reachable so it isn't likely that there will be download issues.